src/axelrod_fortran/player.py

```diff
@@ -1,20 +1,114 @@
+from collections import defaultdict
+from ctypes import cdll, c_int, c_float, byref, POINTER
+from ctypes.util import find_library
+import os
 import random
+import shutil
+import tempfile
+import uuid
 import warnings
 
 import axelrod as axl
 from axelrod.interaction_utils import compute_final_score
 from axelrod.action import Action
-from ctypes import cdll, c_int, c_float, byref, POINTER
 from .strategies import characteristics
 
 C, D = Action.C, Action.D
 actions = {0: C, 1: D}
 original_actions = {C: 0, D: 1}
 
 
+self_interaction_message = """
+You are playing a match with the same player against itself. However
+axelrod_fortran players share memory. You can initialise another instance of an
+Axelrod_fortran player with player.clone().
+"""
+
+
+class LibraryManager(object):
+    """LibraryManager creates and loads copies of a shared library, which
+    enables multiple copies of the same strategy to be run without the end user
+    having to maintain many copies of the shared library.
+
+    This works by making a copy of the shared library file and loading it into
+    memory again. Loading the same file again will return a reference to the
+    same memory addresses.
+
+    Additionally, library manager tracks how many copies of the library have
+    been loaded, and how many copies there are of each Player, so as to load
+    only as many copies of the shared library as needed.
+    """
+
+    def __init__(self, shared_library_name, verbose=False):
+        self.shared_library_name = shared_library_name
+        self.verbose = verbose
+        self.library_copies = []
+        self.player_indices = defaultdict(set)
+        self.player_next = defaultdict(set)
+        # Generate a random prefix for tempfile generation
+        self.prefix = str(uuid.uuid4())
+        self.library_path = self.find_shared_library(shared_library_name)
+
+    def find_shared_library(self, shared_library_name):
+        ## This finds only the relative path to the library, unfortunately.
```

Comment:
I wonder if the `absolute` method of the Pathlib library might be able to help us out here

Reply:
I'll look into it. I think the issue is that the library loader doesn't return the absolute path for some reason (though it clearly knows it) so we may need to either look for the file somehow (in a cross platform way) or instruct the user to symlink the file (not ideal).

Maybe if we have an install script for the shared library here we can capture the path at installation time and write to a config file.